File: core/src/main/kotlin/eldoria/core/world/WorldConfig.kt
package eldoria.core.world

/**
 * width * height must be >= 10,000. Default 130x90 = 11,700 map tiles,
 * split into 6 vertical biome bands (~1,950 tiles each).
 */
data class WorldConfig(
    val width: Int = 130,
    val height: Int = 90,
    val seed: Long = 1337L,
    val citiesPerBiome: Int = 2,
    val villagesPerCity: Int = 4,
    val dungeonsPerBiome: Int = 3,
    val dungeonRoomCountRange: IntRange = 12..20,
    val beanstalksPerBiome: Int = 2,
    val skyRoomCountRange: IntRange = 10..18,
    /** Rivers cutting west-to-east through the 5 land biomes -- impassable on foot except at bridges. */
    val riverCount: Int = 3,
    /** Roughly 1 bridge every this-many tiles along a river's length. */
    val bridgeSpacing: Int = 10,
)

File: eldoria/world/world_config.py
"""width * height must be >= 10,000. Default 130x90 = 11,700 map tiles, 6 vertical biome bands."""
from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True)
class WorldConfig:
    width: int = 130
    height: int = 90
    seed: int = 1337
    cities_per_biome: int = 2
    villages_per_city: int = 4
    dungeons_per_biome: int = 3
    dungeon_room_count_range: range = range(12, 21)
    beanstalks_per_biome: int = 2
    sky_room_count_range: range = range(10, 19)
    river_count: int = 3
    bridge_spacing: int = 10

    def __post_init__(self):
        assert self.width * self.height >= 10_000
